lix/packages/sdk/src/resolve-conflict/resolve-conflict.ts

```diff
@@ -0,0 +1,111 @@
+import type { Change, Conflict } from "../schema.js";
+import type { Lix } from "../types.js";
+import {
+	ChangeDoesNotBelongToFileError,
+	ChangeHasBeenMutatedError,
+	ChangeNotDirectChildOfConflictError,
+} from "./errors.js";
+
+/**
+ * Resolves a conflict by applying the given change.
+ */
+export async function resolveConflict(args: {
+	lix: Lix;
+	conflict: Conflict;
+	resolveWithChange: Change;
```

Comment:
why not prevent the error cases by just allowing a change id here? the creating of a new change in conflict resolution would also likely be better done the same way as normal change creation? (not 100% sure about this)

Reply:
> the creating of a new change in conflict resolution would also likely be better done the same way as normal change creation? (not 100% sure about this)

Rollbacks will be hard/buffy. 

If an app, or even lix itself, needs to create a new change as part of resolving a conflict, marking a conflict as resolved and inserting the change should happen in one transaction. 

I'll leave the API like that for now. We can revisit in the future.